drivers/serial/uart_mcux.c

```diff
@@ -349,6 +351,32 @@ static int uart_mcux_init(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int uart_mcux_pm_action(const struct uart_mcux *dev, enum pm_device_action action)
+{
+	const struct uart_mcux_config *config = dev->config;
+	switch (action)
+	{
+
+		case PM_DEVICE_ACTION_RESUME:
+			// set the UARTSWAI bit to continue normal UART functionality
+			config->base->C1 |= UART_C1_UARTSWAI(1);
+		break;
+		case PM_DEVICE_ACTION_SUSPEND:
+			// clear the UARTSWAI bit to enable low power activity while in wait mode
+			config->base->C1 &= ~UART_C1_UARTSWAI(1);
+		break;
```

Comment:
@mmahadevan108 can you check it please ?

Reply:
Minor issue, please fix the indentation of the break commands

Reply:
Looks like this bit keeps the UART clock running during low power modes. Should this be something the user gets to decide through a device tree property?

Reply:
Another point is that we don't need to clear and enable this bit in suspend and resume. This is a one time operation where the user decides if the UART clock should be active in low power modes.